Content.Server/Chat/V2/Repository/ChatRepository.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Shared.Chat.V2;
+using Content.Shared.Chat.V2.Repository;
+using Robust.Server.Player;
+using Robust.Shared.Replays;
+
+namespace Content.Server.Chat.V2.Repository;
+
+/// <summary>
+/// Stores <see cref="IChatEvent"/>, gives them UIDs, and issues <see cref="MessageCreatedEvent"/>.
+/// Allows for deletion of messages.
+/// </summary>
+public sealed class ChatRepository : EntitySystem
+{
+    [Dependency] private readonly IReplayRecordingManager _replay = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+
+    // Clocks should start at 1, as 0 indicates "clock not set" or "clock forgotten to be set by bad programmer".
+    private uint _nextMessageId = 1;
+    private Dictionary<uint, ChatRecord> _messages = new();
+    private Dictionary<string, Dictionary<uint, ChatRecord>> _playerMessages = new();
```

Comment:
Is there a reason why you're using string instead of uint?

Reply:
_playerMessages is a map of player uid to a map of message IDs to messages

Realistically I could simplify this to just being a map of player UIDs to a list of message IDs though

Reply:
I mean that you're converting all the incoming uint userIds to strings before processing them in this system. Why not just use uint instead of converting to strings?